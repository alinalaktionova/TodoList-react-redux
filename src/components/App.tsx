import React, {Dispatch} from "react";
import TodoList from "./TodoList";
import TodoInput from "./TodoInput";
import {connect} from "react-redux";
import styled from "styled-components";
import PropsInterface from "../interfaces/PropsInterface";
import StateInterface from "../interfaces/StateInterface";
import TodoInterface from "../interfaces/TodoInterface";

const Container = styled.div`
  width: 50%;
  margin: 0 auto;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 13px;
`;

const Header = styled.h1`
  text-transform: capitalize;
  text-align: center;
  text-shadow: 2px 1px 3px black;
  color: dimgrey;
  font-size: 50px;
`;


export const App = (props: Partial<PropsInterface>) => {
  console.dir(props);
  return (
    <Container>
      <Header>ToDo list</Header>
      <TodoInput/>
      <TodoList
        todos={props.todos}
      />
    </Container>
  );
};

const mapStateToProps = (state: StateInterface) => ({
    todos: state.todos
});

export default connect(mapStateToProps)(App);
