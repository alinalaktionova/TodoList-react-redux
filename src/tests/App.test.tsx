import React, {ReactChild, ReactChildren, ReactComponentElement} from "react";
import App from "../components/App";
import { Provider } from "react-redux";
import configureStore, { MockStore } from "redux-mock-store";
import renderer from "react-test-renderer";

const mockStore = configureStore([]);

describe("My Connected React-Redux Component", () => {
  let store: MockStore;
  let component: any;
  beforeEach(() => {
    store = mockStore({ todos: [] });
    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
