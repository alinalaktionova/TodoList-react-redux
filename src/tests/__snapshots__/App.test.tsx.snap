// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="sc-gzVnrw ARDTO"
>
  <h1
    className="sc-htoDjs cZhAiA"
  >
    ToDo list
  </h1>
  <div
    className="sc-ifAKCX hsmLUJ"
  >
    <input
      className="sc-EHOje eqDocV"
      onChange={[Function]}
      placeholder="add a todo item"
      type="text"
    />
    <button
      className="sc-bZQynM hJssYL"
      onClick={[Function]}
      type="button"
    >
      + Add
    </button>
  </div>
  <ul
    className="sc-htpNat lcxLoP"
  >
    <button
      className="sc-bxivhb hAhGGw"
      onClick={[Function]}
      type="button"
    >
      delete all
    </button>
  </ul>
</div>
`;
